### PR TITLE
service: verify instance name and use gibibyte

### PR DESCRIPTION
**Pull request by gozssky**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -24,3 +24,5 @@ anyhow = { version = "1.0" }
 google-signin = { git = "https://github.com/hi-rustin/google-signin-rs" }
 rand = "0.8.4"
 either = "1.6"
+lazy_static = "1.4.0"
```
Comment (hi-rustin) [inline Cargo.toml:27]: Why not just use oncecell's lazy?

File: src/service.rs [hunk 2]
```diff
@@ -19,12 +21,25 @@ use crate::{
 };
 
 pub fn protected_routes() -> Router {
+    // Instance name will be used as kubernetes's resource names, such as pod names, label names,
+    // hostnames and so on. So the same naming constraints should be applied to the instance name.
+    // See: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names.
+    fn verify_instance_name(name: &str) -> bool {
```
Comment (hi-rustin) [inline src/service.rs:27]: Maybe it should be put outside? But that seems to work too.

File: src/service.rs [hunk 3]
```diff
@@ -18,13 +20,24 @@ use crate::{
     model::{Instance, InstanceStage},
 };
 
+/// Returns true if and only if the name is a valid instance name.
+///
+/// Instance name will be used as kubernetes's resource names, such as pod names, label names,
+/// hostnames and so on. So the same naming constraints should be applied to the instance name.
+/// See: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names.
+fn verify_instance_name(name: &str) -> bool {
+    static RE: Lazy<Regex> =
```
Comment (hi-rustin) [inline src/service.rs:29]: Maybe it could be put outside? 😆  

We have the test!

**Review by hi-rustin — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (gozssky) [reply Cargo.toml]: Just copy the example of [regex](https://github.com/rust-lang/regex#usage-avoid-compiling-the-same-regex-in-a-loop) 🤣, seems oncecell's lazy is also ok here. I have updated it.

File: src/service.rs (hunk 3 above)
Comment (gozssky) [reply src/service.rs:29]: make sense

**Review by hi-rustin — APPROVED** [review]

**Pull request merged**